test/compile_test/test_compile_result.hpp

```diff
@@ -52,7 +52,10 @@ template <class T1, class T2>
 inline void check_result_imp(T1, T2)
 {
    // This is a static assertion that should always fail to compile...
+#pragma GCC diagnostic push
```

Comment:
Do we know which GCC version first started supporting this?

Reply:
4.8: https://gcc.gnu.org/bugzilla/show_bug.cgi?id=33255